pkg/podman/podman.go

```diff
@@ -144,6 +146,36 @@ func (p *Podman) Copy(id, src, dest string) error {
 	return nil
 }
 
+// Inspect returns the sha256 digest for a given container image on a specific architecture
+func (p *Podman) Inspect(img string, arch string) (string, error) {
+	zap.S().Infof("Inspecting %s for linux/%s", img, arch)
+
+	options := new(manifests.InspectOptions)
+	m, err := manifests.Inspect(p.context, img, options)
+	if err != nil {
+		return "", fmt.Errorf("error inspecting manifest %s: %w", img, err)
+	}
+
+	for i := range m.Manifests {
+		if m.Manifests[i].Platform.OS == "linux" && m.Manifests[i].Platform.Architecture == arch {
+			foundDigest := m.Manifests[i].Digest.String()
+
+			// This is done to remove "sha:" from the digest
+			digest := foundDigest
+			if strings.Contains(foundDigest, ":") {
+				parts := strings.Split(foundDigest, ":")
+				if len(parts) == 2 {
+					digest = parts[1]
+				}
+			}
+
+			return digest, nil
+		}
```

Comment:
Simplifying suggestion:
```suggestion
	for _, manifest := range m.Manifests {
		if manifest.Platform.OS == "linux" && manifest.Platform.Architecture == arch {
			digest := manifest.Digest.String()

			// Remove "sha:" from the digest if present
			if parts := strings.Split(digest, "sha:"); len(parts) == 2 {
				digest = parts[1]
			}

			return digest, nil
		}
	}
```

Reply:
This is done because the linter complains about 
```
pkg/combustion/registry.go:289:2: rangeValCopy: each iteration copies 176 bytes (consider pointers or indexing) (gocritic)
```